api/envoy/config/filter/network/ext_authz/v2/ext_authz.proto

```diff
@@ -43,4 +44,11 @@ message ExtAuthz {
   // When this field is true, Envoy will include the peer X.509 certificate, if available, in the
   // :ref:`certificate<envoy_api_field_service.auth.v2.AttributeContext.Peer.certificate>`.
   bool include_peer_certificate = 4;
+
+  // Specifies if HTTP status codes for gRPC calls reflect gRPC status
+  // https://github.com/envoyproxy/envoy/issues/11079
+  //
+  // When this field is true, Envoy will send an HTTP response code appropriate to
+  // the response seen
+  bool http_status_for_grpc_deny = 5;
```

Comment:
No changes to v2 are allowed, these APIs are frozen (Envoy will drop support completely for v2 after the 1.18.0 release going out today).